### PR TITLE
Site Editor: Fix _wp_file_based term deletion in migration

### LINKED ISSUE #28155: Database migration didn't work properly
**Describe the bug**
The database migration introduced in #27910 did not erase all of the data it was designed to erase.

* Auto-drafts are correctly erased but their term relationships didn't get erased. In other words, you end up with "orphan" entries in the term relationships table.

* The `_wp_file_based` term didn't get erased.

* The `gutenberg_last_synchronize_theme_template_checks` option didn't get erased.

* The `gutenberg_last_synchronize_theme_template-part_checks` option didn't get erased.

**Expected behavior**
Either the migration needs to be complete, or don't bother including a migration at all.

**Editor version (please complete the following information):**
- WordPress version: **5.6**
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? **plugin**
- If the Gutenberg plugin is installed, which version is it? **13 Jan 2021 commit on master branch**

### PR DESCRIPTION
**Pull request by ockham**
## Description
Purports to fix #28155 :crossed_fingers: 

> The database migration introduced in #27910 did not erase all of the data it was designed to erase.
> * Auto-drafts are correctly erased but their term relationships didn't get erased. In other words, you end up with "orphan" entries in the term relationships table.
> * The `_wp_file_based` term didn't get erased.

The reason for this is twofold:

1. Execution order: The deletion was attempted from the migration, which was hooked into [`plugins_loaded`](https://developer.wordpress.org/reference/hooks/plugins_loaded/). However, the `wp_theme` taxonomy was only registered upon `init` (see https://github.com/WordPress/gutenberg/issues/28155#issuecomment-762440743).
2. A simple typo: The database migration was looking for the `wp-theme` (dash) rather than the `wp_theme` (underscore) taxonomy (see https://github.com/WordPress/gutenberg/issues/28155#issuecomment-762401534).

## How has this been tested?

Make sure you have a `_wp_file_based` term (in the `wp_theme` taxonomy) lying around from a run of the Site Editor prior to GB 9.8. To verify that you do have one, run

```
npm run wp-env run cli "wp term get wp_theme _wp_file_based --by=slug"
```

(If this comes back empty, downgrade your version of Gutenberg to 9.7 (e.g. `git checkout release/9.7 && npm run build`), install and activate an FSE-enabled theme, and load the Site Editor. Run the above WP CLI command again to verify that it worked.)

Assuming that that object is present:

- Check out this PR's branch, and build it.
- Make sure to delete the `_GUTENBERG_VERSION_MIGRATION` option in order to have the migration triggered: `npm run wp-env run cli "wp option delete _GUTENBERG_VERSION_MIGRATION"`
- Run your local WP install, and access any given page. (This will trigger the `init` hook, and thus, the migration.)
- Verify that the `_wp_file_based` term is gone by once again running:

```
npm run wp-env run cli "wp term get wp_theme _wp_file_based --by=slug"
```

## Types of changes
Bug fix

cc/ @carlomanf for review :slightly_smiling_face:

### CODE REVIEW

**Comment by carlomanf** [comment]
Worked for me. Is there any chance that `_GUTENBERG_VERSION_MIGRATION` can be changed to `gutenberg_version_migration` just to match the usual formatting convention for options entries?

File: lib/upgrade.php [hunk 1]
```diff
@@ -49,14 +49,17 @@ function _gutenberg_migrate_remove_fse_drafts() {
 	}
 
 	// Delete _wp_file_based term.
-	$term = get_term_by( 'name', '_wp_file_based', 'wp-theme' );
+	$term = get_term_by( 'name', '_wp_file_based', 'wp_theme' );
```
Comment (youknowriad) [inline lib/upgrade.php:52]: 😅

**Review by youknowriad — APPROVED** [review]

**Comment by ockham** [comment]
> Worked for me.

Thanks for testing!

> Is there any chance that `_GUTENBERG_VERSION_MIGRATION` can be changed to `gutenberg_version_migration` just to match the usual formatting convention for options entries?

I like the idea -- it also would make it less easy to confuse the option with the eponymous code constant.

@youknowriad Would you be okay with that?

**Comment by youknowriad** [comment]
> Is there any chance that _GUTENBERG_VERSION_MIGRATION can be changed to gutenberg_version_migration just to match the usual formatting convention for options entries?

Makes sense, but we should keep the const as is.

**Comment by ockham** [comment]
Cherry-picked to `release/9.8`.

**Pull request merged**